docs/6-infrastructure-configuration-management/6.0-overview.md

```diff
@@ -0,0 +1,19 @@
+# 6.0 Infrastructure and Configuration Management
+
+## Chapter Goals
+
+ - Learn how different infrastructure and configuration management tools work and what features they support
+ - Understand the challenges that enterprise organizations face managing their infrastructure and configuring their environments.
+ - Consider how the concept of immutable infrastructure relates to the other concepts in this chapter
+ - Learn the basics of writing Terraform modules and Ansible playbooks
+
+![](../img/goals_light.svg ':size=100x100 :class=light-mode-icon')
+![](../img/goals_dark.svg ':size=100x100 :class=dark-mode-icon')
+
+## Concepts
+
+Managing infrastructure generally involves provisioning new resources and configuring those resources. Sometimes the distinction between these two operations is blurred but thinking about how (or if) the tools in the chapter handle each can help understand why the tools are designed the way they are. Your experience in chapter *3 Cloud Computing* gave you some examples for provisioning new resources and you should have come across some examples of configuring systems in the virtualization part of chapter *2 Virtual Machines and Containers*
+
+### Immutable Infrastructure
+
+Many older configuration management tools focus on the idea of applying configuring changes in a uniform way to a set of servers or virtual machines. This of course goes directly against the more modern idea of immutable infrastructure. However there are many instances where older environments use these tools to configure machines or edge cases where immutable infrastructure may not be desirable.
```

Comment:
"Many older configuration management tools focus on the idea of applying configuring changes in a uniform way to a set of servers or virtual machines. This of course goes directly against the more modern idea of immutable infrastructure. However[,] there are many instances where older environments use these tools to configure machines or edge cases where immutable infrastructure may not be desirable."